h/api/auth.py

```diff
@@ -13,3 +15,21 @@ def get_user(request):
                 consumer = Consumer(key)
                 return User(userid, consumer, False)
     return None
+
+
+def translate_annotation_principals(principals):
+    """
+    Translate a list of annotation principals to a list of pyramid principals.
+    """
+    result = set([])
+    for principal in principals:
+        # Ignore suspicious principals from annotations
+        if principal.startswith('system.'):
```

Comment:
The problem is that `translate_annotation_principals` could be used in many different contexts, and I can't predict what all of those will be. Allowing data from outside to map to pyramid "special" principals is an accident waiting to happen.